src/App/Version.hs

```diff
@@ -13,26 +13,29 @@ import Data.String.Conversion (toText)
 import Data.Text (Text)
 import Data.Text qualified as Text
 import Data.Version (showVersion)
-import GitHash (GitInfo, giBranch, giDirty, giHash, tGitInfoCwd)
+import GitHash (GitInfo, giBranch, giDirty, giHash, tGitInfoCwdTry)
 import System.Info (compilerName, compilerVersion)
 
 versionNumber :: Maybe Text
 versionNumber = $$(getCurrentTag)
 
-info :: GitInfo
-info = $$(tGitInfoCwd)
+info :: Either String GitInfo
+info = $$(tGitInfoCwdTry)
+
+fromRight' :: v -> (r -> v) -> Either l r -> v
+fromRight' v = either (const v)
 
 currentBranch :: Text
-currentBranch = toText $ giBranch info
+currentBranch = toText $ fromRight' "unknown" giBranch info
```

Comment:
Nit: I wonder if these might read better as `toText . fromRight "unknown" $ fmap gitBranch info` or something, just to avoid a non-standard `fromRight'`.  I don't really have a good solution, but it feels like a pattern that should exist without the need for `fromRight'`.

Reply:
If you're opposed to `fromRight'`, I can just use `either` directly. I figured it was a little clearer with the repeated helper factored out.